analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContextBase.cs

```diff
@@ -30,13 +30,9 @@ public class SonarAnalysisContextBase
     protected static readonly ConditionalWeakTable<Compilation, ConcurrentDictionary<string, bool>> FileInclusionCache = new();
     protected static readonly ConditionalWeakTable<Compilation, ImmutableHashSet<string>> UnchangedFilesCache = new();
     protected static readonly SourceTextValueProvider<ProjectConfigReader> ProjectConfigProvider = new(x => new ProjectConfigReader(x));
```

Comment:
Verifying: Do these `SourceTextProviders` pick everything that is picked up by the compilation and try to process based on the generic we provide (with `TryGetValue`)?

Reply:
I don't understand the question

Reply:
When we define `SourceTextValueProvider<T>` does the provider check every file that is included in the compilation?

Reply:
No. It only processes files that we explicitly throw at it. See

https://github.com/SonarSource/sonar-dotnet/blob/d73c2cce42f48ca4f91c6003a2186157292bf5be/analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContextBase.cs#L70-L74

Where `Options.SonarProjectConfig()` looks into additional files and chooses the first with `SonarProjectConfig.xml` name. And only this one is thrown on the `TryGetValue`, that will eventually end up in the `SourceTextValueProvider` on the 1st run (cache miss run)